modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserLocalServiceTest.java

```diff
@@ -498,130 +728,70 @@ public void testLockoutUser() throws Exception {
+						passwordPolicy.setLockout(true);
+						passwordPolicy.setMaxFailure(1);
+					})) {
 
-		try {
-			_userLocalService.authenticateByEmailAddress(
-				user.getCompanyId(), user.getEmailAddress(), password, null,
-				null, null);
-		}
-		catch (PortalException portalException) {
-			Assert.assertEquals(
-				UserLockoutException.PasswordPolicyLockout.class,
-				portalException.getClass());
-		}
+			int failedLoginAttempts = user.getFailedLoginAttempts();
 
-		try {
-			_userLocalService.authenticateByEmailAddress(
-				user.getCompanyId(), user.getEmailAddress(),
-				RandomTestUtil.randomString(), null, null, null);
-		}
-		catch (PortalException portalException) {
 			Assert.assertEquals(
-				AuthException.class, portalException.getClass());
-		}
-
-		user = _userLocalService.fetchUser(user.getUserId());
-
-		Assert.assertEquals(
-			failedLoginAttempts + 3, user.getFailedLoginAttempts());
-
-		passwordPolicy = user.getPasswordPolicy();
-
-		passwordPolicy.setLockout(false);
-
-		_passwordPolicyLocalService.updatePasswordPolicy(passwordPolicy);
-
-		Assert.assertEquals(
-			Authenticator.SUCCESS,
-			_userLocalService.authenticateByEmailAddress(
-				user.getCompanyId(), user.getEmailAddress(), password, null,
-				null, null));
-	}
-
-	@Test
-	public void testPasswordHistory() throws Exception {
-		User user = UserTestUtil.addUser();
-
-		PasswordPolicy passwordPolicy = user.getPasswordPolicy();
-
-		passwordPolicy.setHistory(true);
-		passwordPolicy.setHistoryCount(2);
-
-		_passwordPolicyLocalService.updatePasswordPolicy(passwordPolicy);
-
-		String password1 = "password1";
-		String password2 = "password2";
-
-		try {
-			ServiceContextThreadLocal.pushServiceContext(
-				ServiceContextTestUtil.getServiceContext(
-					user.getGroupId(), user.getUserId()));
-
-			user = _userLocalService.updatePassword(
-				user.getUserId(), password1, password1, false, false);
+				Authenticator.FAILURE,
+				_userLocalService.authenticateByEmailAddress(
+					user.getCompanyId(), user.getEmailAddress(),
+					RandomTestUtil.randomString(), null, null, null));
 
-			user = _userLocalService.updatePassword(
-				user.getUserId(), password2, password2, false, false);
+			try {
+				_userLocalService.authenticateByEmailAddress(
+					user.getCompanyId(), user.getEmailAddress(), password, null,
+					null, null);
+			}
+			catch (PortalException portalException) {
+				Assert.assertEquals(
+					UserLockoutException.PasswordPolicyLockout.class,
+					portalException.getClass());
+			}
 
-			Assert.assertEquals(
-				Authenticator.SUCCESS,
+			try {
 				_userLocalService.authenticateByEmailAddress(
-					user.getCompanyId(), user.getEmailAddress(), password2,
-					null, null, null));
+					user.getCompanyId(), user.getEmailAddress(),
+					RandomTestUtil.randomString(), null, null, null);
+			}
+			catch (PortalException portalException) {
+				Assert.assertEquals(
+					AuthException.class, portalException.getClass());
+			}
```

Comment:
User AssertUtils.assetFailure consistently.

Reply:
I removed this, because we no longer throw an exception, but rather return Authenticator.FAILURE (-1).  Since we already performed this exact check earlier, I've removed it.